### PR TITLE
[17.0] [MIG] stock_available: Migration to 17.0

### PR DESCRIPTION
**Pull request by bizzappdev**

### CODE REVIEW

**Review by rousseldenis — APPROVED** [review]
Code review.

Even if I suggest to move the 'mrp' configuration in the mrp available related module.

**Comment by bizzappdev** [comment]
> Code review.
> 
> Even if I suggest to move the 'mrp' configuration in the mrp available related module.

I'm not sure If I understand you correctly, but If we move the MRP configuration to the available MRP module, then that configuration will not be useful. This configuration itself is installing the module.

**Comment by pedrobaeza** [comment]
For me, that configuration should go out, as it's very poisoned: if the module is not migrated, it's not working and do users to put issues about it. And even if the module is migrated, I don't like the checks that install modules without total user awareness.

**Comment by bizzappdev** [comment]
> For me, that configuration should go out, as it's very poisoned: if the module is not migrated, it's not working and do users to put issues about it. And even if the module is migrated, I don't like the checks that install modules without total user awareness.

I concur with your point @pedrobaeza . Let me remove it.
Also there is a configuration for installing stock_available_immediately, That should also go away IMO.

**Comment by bizzappdev** [comment]
> For me, that configuration should go out, as it's very poisoned: if the module is not migrated, it's not working and do users to put issues about it. And even if the module is migrated, I don't like the checks that install modules without total user awareness.

@pedrobaeza @rousseldenis  Removed the configuration for module installation.

File: stock_available/data/ir_config_parameter.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<odoo noupdate="1">
+    <record id="default_stock_available_mrp_based_on" model="ir.config_parameter">
+        <field name="key">stock_available.stock_available_mrp_based_on</field>
```
Comment (pedrobaeza) [inline stock_available/data/ir_config_parameter.xml]: Why having this parameter for `mrp`?

File: stock_available/views/res_config_settings_views.xml [hunk 2]
```diff
@@ -8,38 +8,17 @@
...
-                                        attrs="{'required':[('module_stock_available_mrp','=',True)]}"
-                                    />
-                                </div>
+            <xpath expr="//block[@id='schedule_info']" position="after">
+                <block title="Stock available to promise" name="available_info">
+                    <setting>
+                        <label for="stock_available_mrp_based_on" />
+                        <div class="content-group">
+                            <div class="mt16">
+                                <field name="stock_available_mrp_based_on" />
```
Comment (pedrobaeza) [inline stock_available/views/res_config_settings_views.xml]: This shouldn't be here IMO.

File: stock_available/data/ir_config_parameter.xml (hunk 1 above)
Comment (bizzappdev) [reply stock_available/data/ir_config_parameter.xml]: Basically, this is also a field for configuration, and it is only used for the MRP bridge module.
Previously, it was with get_values and set_values with default value.
We have changed it to the config_parameter and created the data for it to support the default value.

It is also a question of whether it should remain in this module or be moved to the MRP bridge module.

Please suggest the proper approach, and we will do the same.

Comment (pedrobaeza) [reply stock_available/data/ir_config_parameter.xml]: But it's the same as the module installation: this is for modules outside this one, so they should be on the corresponding module (`stock_available_mrp`).

Comment (bizzappdev) [reply stock_available/data/ir_config_parameter.xml]: Done. We will remember when migrating stock_available_mrp to include that.

**Comment by pedrobaeza** [comment]
@rousseldenis do you see correct to remove that field? Or maybe I'm misunderstanding the meaning of it?

**Comment by rousseldenis** [comment]
Yes, that seems fair for me.

**Review by pedrobaeza — APPROVED** [review]

**Review by Abranes — APPROVED** [review]

**Comment by rpinset** [comment]
Hello, could there not be a calculation problem?
The immediately_usable_qty field gives the forecast quantity taking into account supplier receipts not yet received.
image

This does not match the field name and description
https://github.com/OCA/stock-logistics-availability/blob/ac926a5e3178369d8b252722622e8ab0c3bfae57/stock_available/models/product_product.py#L56

File: stock_available/data/ir_config_parameter.xml (hunk 1 above)
Comment (rpinset) [reply stock_available/data/ir_config_parameter.xml]: @pedrobaeza an opinion regarding this question 

https://github.com/BizzAppDev-Systems/stock-logistics-availability/pull/2/files#r1661123966
> In v16, the ICP was in the stock_available module. Now, it has been moved to the stock_available_mrp module. This migration script will be helpful for those who already have the v16 version installed for them. Question is: For the migration script, should we keep the old name and move it to this module or change the name?

IMO we need to change the name of the existing key to v16.0 when migrating to v17.0

**Pull request merged**